### PR TITLE
add setting for project opening method

### PR DESCRIPTION
**Pull request by Eskibear**
for #662

### CODE REVIEW

**Comment by jdneo** [comment]
Ci fails

File: package.json [hunk 1]
```diff
@@ -673,6 +673,17 @@
             "default": "true",
             "description": "%configuration.maven.dependency.enableConflictDiagnostics%",
             "scope": "window"
+          },
+          "maven.defaultOpenProjectMethod": {
```
Comment (jdneo) [inline package.json:677]: maven.projectOpenBehavior

File: package.json [hunk 2]
```diff
@@ -673,6 +673,17 @@
...
+            "default": "Interactive",
```
Comment (jdneo) [inline package.json:678]: Do we have data for the favorite option? I prefer `Open` personally

File: package.nls.json [hunk 3]
```diff
@@ -38,5 +38,6 @@
     "configuration.maven.terminal.favorites.command": "Content of the favorite command.",
     "configuration.maven.terminal.favorites.debug": "Whether to execute in debug mode.",
     "configuration.maven.settingsFile": "Specifies the absolute path of your maven configuration file, the default value is ~/.m2/settings.xml",
-    "configuration.maven.dependency.enableConflictDiagnostics": "Specify whether to show diagnostics for conflict dependencies."
+    "configuration.maven.dependency.enableConflictDiagnostics": "Specify whether to show diagnostics for conflict dependencies.",
+    "configuration.maven.defaultOpenProjectMethod": "Default method of opening newly created project."
```
Comment (jdneo) [inline package.nls.json:42]: Default behavior xxx...

File: src/extension.ts [hunk 4]
```diff
@@ -256,26 +256,32 @@ async function openPomHandler(node: MavenProject | { uri: string }): Promise<voi
 }
 
 function registerProjectCreationEndListener(context: vscode.ExtensionContext): void {
+    const OPEN_IN_NEW_WORKSPACE = "Open";
+    const OPEN_IN_CURRENT_WORKSPACE = "Add to Workspace";
+
+    const specifyOpenMethod = async (hasOpenFolder: boolean, projectName: string, projectLocation: string) => {
+        let openMethod = vscode.workspace.getConfiguration("maven").get<string>("defaultOpenProjectMethod");
```
Comment (jdneo) [inline src/extension.ts:263]: openMethod -> openBehavior

Remember to update other occurrences as well.

File: package.json (hunk 1 above)
Comment (Eskibear) [reply package.json:677]: I directly followed spring-initializr, see https://github.com/microsoft/vscode-spring-initializr/pull/157

A better name in my mind was `maven.project.postCreationBehavior"

File: package.json (hunk 2 above)
Comment (Eskibear) [reply package.json:678]: > Do we have data for the favorite option?

No we don't. We can add corresponding telemetry. Then we need to keep "interactive" as default for a while to collect data.

I'm 50:50, maybe prefer interactive a little bit. maybe my key point is, I don't think this notification is a frequent thing.

File: package.json (hunk 1 above)
Comment (jdneo) [reply package.json:677]: I'm referring from https://github.com/microsoft/vscode-azurefunctions/blob/54c7614fbb984fbf3ff6bff1028979bd96f3a644/package.json#L952

postCreation sounds like a lifecycle method name but not a setting name

File: package.json (hunk 2 above)
Comment (jdneo) [reply package.json:678]: Colleting data sounds good

File: package.json (hunk 1 above)
Comment (Eskibear) [reply package.json]: then let's follow azfunc

File: package.json (hunk 2 above)
Comment (Eskibear) [reply package.json:678]: will sendInfo

**Comment by Eskibear** [comment]
telemetry schema: 

```
info {
  name: proejectOpenBehavior(from setting/choice)
  value: open/add-to-workspace/cancelled
}
```
@jdneo see if it works

**Review by jdneo — APPROVED** [review]

**Comment by Eskibear** [comment]
pending event to tag:   `info { value: string }`

**Pull request merged**